File: API/Features/Reservations/Bookings/Dtos/BookingReadDto.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations.Bookings {

    public class BookingReadDto : IMetadata {

        // PK
        public Guid BookingId { get; set; }
        // FKs
        public int BoatTypeId { get; set; }
        // Fields
        public string BoatName { get; set; }
        public decimal Length { get; set; }
        public decimal Width { get; set; }
        public string FromDate { get; set; }
        public string ToDate { get; set; }
        public int Days { get; set; }
        public string ContactDetails { get; set; }
        public string Email { get; set; }
        public string Remarks { get; set; }
        public bool IsConfirmed { get; set; }
        public List<BookingPierVM> Piers { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/bookings/classes/dtos/booking-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface BookingReadDto extends Metadata {

    bookingId: Guid
    boatType: SimpleEntity
    boatName: string
    boatLength: string
    fromDate: string
    toDate: string
    stayDuration: number
    contactDetails: string
    email: string
    remarks: string
    isConfirmed: boolean
    isDocked: boolean
    isPaid: boolean
    postAt: string
    postUser: string
    putAt: string
    putUser: string

}
